src/imp/run.c

```diff
@@ -281,6 +281,7 @@ int imp_run_privileged (struct imp_state *imp,
     struct kv *kv_env;
     const char *name;
     const cf_t *cf_run;
+    const char *notify_socket = getenv ("NOTIFY_SOCKET");
```

Comment:
Could instead add this variable to `run_env_var_allowed()` for consistency with `FLUX_JOB_ID` and `FLUX_JOB_USERID`.

If that isn't desirable, a comment here might help remind a reader why this variable is special (though admittedly at this point it's obvious)

Reply:
Hmm, i tried that originally and it didn't work for some reason and got myself a little confused in the process. Let me try again and see if I can run down what's happening.

Reply:
Oh, the privileged imp process has its environment cleared, and the environment curated by the unprivileged imp process is passed through to `execve()`. But the imp itself, whose environment remains empty, will be making the `sd_notify()` calls so this does need to be handled differently.

I'll add a comment.